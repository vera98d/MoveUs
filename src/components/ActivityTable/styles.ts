import styled from "styled-components";

export const TableWrapper = styled.div`
  width: 80vh;
  height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ExerciseTable = styled.div`
  display: grid;
  text-align: center;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: [row1-start] 15%;
  column-rule: 1px solid red;
  grid-auto-rows: 12.14%;
  border: 1px solid black;
  width: 100%;
  height: 50vh;
`;

export const ExerciseCell = styled.p`
  padding: 8px 4px;
  border-left: 1px solid black;
  border-bottom: 1px solid black;
`;

export const PaginationWrapper = styled.div`
  width: 80%;
  display: flex;
  justify-content: space-around;
`;

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;
  
  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ListElement = styled.li`
`;
