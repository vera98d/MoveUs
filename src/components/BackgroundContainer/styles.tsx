import styled from "styled-components";

const BackgroundContainer = styled.div`
    width: 100%;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 2em 3em;
    background: ${(props) => props.theme.colors.background};
`;

export default BackgroundContainer;