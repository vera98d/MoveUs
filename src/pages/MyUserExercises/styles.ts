import styled from "styled-components";

export const BackgroundContainer = styled.div`
  width: 100%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 2em 3em;
  background: ${(props) => props.theme.colors.background};
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;
  
  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 45px;
  }
  
  .setPages {
    margin-bottom: 50px;
  }
  
  h3 {
    margin-bottom: 30px;
  }
  
  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  
  .usernameStyle {
    color: ${(props) => props.theme.colors.secondaryUi};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 18px;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
  
  .lineWrapperContent {
    margin: 20px 0;
    width: 100%;
    display: flex;
    justify-content: space-between;
  }
`;

export const GroupContainer = styled.div`
  margin-bottom: 60px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  
  .verticalLine {
    padding: 0 15px;
  }
`;
