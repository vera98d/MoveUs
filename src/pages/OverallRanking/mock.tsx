import { User } from "../../interfaces/dbData";

export const users: User[] = [
  {
    uid: "1",
    name: "John",
    surname: "Mayer",
    login: "JohnMayer",
    password: "passwd17654",
    email: "johnmayer@gmail.com",
    score: 1340,
    lastActivity: new Date(2022, 4, 4),
    groups: ["1", "2", "4", "5"],
    ownedGroups: ["1", "5"],
    activities: ["1", "2"],
  },
  {
    uid: "2",
    name: "Alicja",
    surname: "Keys",
    login: "AlicjaKeys",
    password: "fwswd17654",
    email: "AlicjaKeys@gmail.com",
    score: 640,
    lastActivity: new Date(2022, 2, 3),
    groups: ["1", "2"],
    ownedGroups: ["2"],
    activities: ["1", "2"],
  },
  {
    uid: "3",
    name: "Billie",
    surname: "Eilish",
    login: "BillieEilish",
    password: "fwlhdik17654",
    email: "BillieEilish@gmail.com",
    score: 100,
    lastActivity: new Date(2022, 2, 10),
    groups: ["1", "3", "4"],
    ownedGroups: ["3"],
    activities: ["3", "2", "4"],
  },
  {
    uid: "4",
    name: "Harry",
    surname: "Styles",
    login: "Harry864",
    password: "fllpjncdik176421",
    email: "BillieEilish@gmail.com",
    score: 940,
    lastActivity: new Date(2022, 4, 8),
    groups: ["1", "2", "4", "5"],
    ownedGroups: ["4"],
    activities: ["3", "2", "4"],
  },
  {
    uid: "5",
    name: "Paul",
    surname: "Hewson",
    login: "Bono123",
    password: "lrootok176421",
    email: "Bono123@gmail.com",
    score: 1540,
    lastActivity: new Date(2022, 4, 15),
    groups: ["1", "4"],
    ownedGroups: [],
    activities: ["3", "2", "4"],
  },
  {
    uid: "6",
    name: "John",
    surname: "Doe",
    login: "jo-do",
    password: "shdjdhjshjs",
    email: "johndoe@gmail.com",
    score: 2150,
    lastActivity: new Date(2022, 4, 12),
    groups: [],
    ownedGroups: [],
    activities: [],
  },
];

export const user: User = users[0];
