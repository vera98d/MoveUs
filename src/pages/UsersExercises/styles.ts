import styled from "styled-components";

export const Img = styled.img`
  border-radius: 50%;
  width: 70px;
  height: 70px;
  margin: 0 30px 0 0;
`;

export const BackgroundContainer = styled.div`
  width: 100%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 2em 3em;
  background: ${(props) => props.theme.colors.background};
`;

export const Header = styled.h1`
  font-size: ${(props) => props.theme.fontSize.header};
  color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  margin-bottom: 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;

  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  h3 {
    margin-bottom: 30px;
  }

  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }

  .usernameStyle {
    color: ${(props) => props.theme.colors.secondaryUi};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 24px;
    font-weight: bold;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
`;
