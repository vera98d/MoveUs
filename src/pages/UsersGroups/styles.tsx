import styled from "styled-components";

interface TileProps {
  img?: string;
}

export const Tile = styled.div<TileProps>`
    width: 130px;
    height: 130px;
    position: relative ;    
    background: ${(props) => (props.img ? `url(${props.img})` : props.theme.colors.white)};
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    opacity: 80%;
    border: ${(props) => props.theme.border};
    border-radius: 25px;
    cursor: pointer;

    &::before{
        content: "";
        display: block;
        position: absolute;
        top: 0;
        left:0;
        width:100%;
        height: 100%;
        background: rgba(255,255,255,0.2);
        border-radius: 25px;
        backdrop-filter: blur(.5px);
    }
`;

export const TileContent = styled.p`
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: ${(props) => props.theme.fontSize.tiles};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    color: ${(props) => props.theme.colors.secondaryUi};
    position: relative ;
    width: 100%;
    height: 100%;
    padding: .5em;
    z-index: 2;
`;
export const TileContainer = styled.div`
    max-width: 1100px;
    width: 100%;
    display: flex;
    align-items:center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 1.5em;
    margin: 1.5em 0;
    text-align: center;
    
    &>a{
        text-decoration: none;
    }
`;
