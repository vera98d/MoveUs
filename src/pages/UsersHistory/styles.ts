import styled from "styled-components";

export const Img = styled.img`
  border-radius: 50%;
  width: 75px;
  height: 75px;
  margin: 0 30px 0 0;
`;

export const Test = styled.div`
`;

export const Header = styled.header`
  height: 130px;
  background-color: aqua;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;
  
  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 45px;
  }
    
  h3 {
    margin-bottom: 30px;
  }
  
  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
`;
