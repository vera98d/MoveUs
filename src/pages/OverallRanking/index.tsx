import RankingTable from "../../components/RankingTable";
import { User } from "../../interfaces/dbData";
import { useEffect, useState } from "react";
import BackgroundContainer from "../../components/BackgroundContainer/styles";
import { Header } from "../../components/Typography/styles";
import userService from "../../services/userService";

const OverallRanking = (): JSX.Element => {
  const [users, setUsers] = useState<User[]>([]);

  useEffect(() => {
    userService.getAllUsers()
      .then((data) => {
        setUsers(data);
      })
      .catch((err: unknown) => {
        if (err instanceof Error) {
          alert(err.message);
        }
      });
  }, []);

  return (
    <BackgroundContainer>
      <Header>
        Overall Ranking
      </Header>
      <RankingTable groupUsers={users} />
    </BackgroundContainer>
  );
};

export default OverallRanking;