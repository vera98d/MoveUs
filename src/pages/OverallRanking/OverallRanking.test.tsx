import { render, screen } from "@testing-library/react";
import {BrowserRouter as Router} from 'react-router-dom';
import { ThemeProvider } from "styled-components";
import OverallRanking from ".";
import userService from "../../services/userService";
import { theme } from "../../styles";
import { users } from "./mock";

jest.mock("firebase/auth", () => ({
  getAuth: () => {}
}));

jest.mock("react-firebase-hooks/auth", () => ({
  useAuthState: () => ([])
}));

let orginalGetAll: any;
beforeEach(() => {
  orginalGetAll = userService.getAllUsers;
  userService.getAllUsers = () => Promise.resolve(users);
});

afterEach(() => {
    userService.getAllUsers = orginalGetAll;
});

describe("OverallRanking", () => {
    test("matches snapshot", async () => {
      const { container } = render(
        <ThemeProvider theme={theme}>
          <Router>
            <OverallRanking />
          </Router>
        </ThemeProvider>
      );
      expect(container).toMatchSnapshot();
    });

    test("is rendering all users", async () => {
        render(
            <ThemeProvider theme={theme}>
              <Router>
                <OverallRanking />
              </Router>
            </ThemeProvider>
          );
        expect(users.length).toBe(users.length);
    });

    test('Header should show "Overall Ranking"', () => {
        render(
          <ThemeProvider theme={theme}>
            <Router>
                <OverallRanking />
              </Router>
          </ThemeProvider>
        );    
        const header = screen.queryByText("Overall Ranking");
        expect(header).toBeInTheDocument();
      });
});
  