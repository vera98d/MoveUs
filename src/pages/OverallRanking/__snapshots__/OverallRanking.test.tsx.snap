// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverallRanking matches snapshot 1`] = `
<div>
  <div
    class="sc-jqUVSM ibeUEl"
  >
    <h1
      class="sc-kDDrLX dwkZS"
    >
      Overall Ranking
    </h1>
    <div
      class="sc-bczRLJ ccrZxC"
    >
      <div
        class="sc-gsnTZi ehtIOX"
      >
        <div
          class="sc-dkzDqf sc-eCYdqJ huYvCz eXzVjt"
        >
          Rank position
        </div>
        <div
          class="sc-dkzDqf sc-eCYdqJ huYvCz eXzVjt"
        >
          Full name
        </div>
        <div
          class="sc-dkzDqf sc-eCYdqJ huYvCz eXzVjt"
        >
          Score
        </div>
        <div
          class="sc-dkzDqf sc-eCYdqJ huYvCz eXzVjt"
        >
          Last activity
        </div>
      </div>
      <div
        class="sc-gsnTZi sc-gKXOVf ehtIOX TGFbt"
      >
        <div
          class="sc-dkzDqf huYvCz"
        />
        <div
          class="sc-dkzDqf huYvCz"
        />
        <div
          class="sc-dkzDqf huYvCz"
        />
        <div
          class="sc-dkzDqf huYvCz"
        />
      </div>
    </div>
  </div>
</div>
`;
